library/Zend/Validator/Date.php

```diff
@@ -101,6 +107,24 @@ public function setFormat($format = self::FORMAT_DEFAULT)
     }
 
     /**
+     * @param boolean $allowNull
+     * @return $this
+     */
+    public function setAllowNull($allowNull)
+    {
+        $this->allowNull = $allowNull;
+        return $this;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function allowNull()
+    {
+        return $this->allowNull;
+    }
+
```

Comment:
The name of this method isn't correct. It should be getAllowNull() because of the way options are set into validator. \Zend\Validator\AbstractValidator::setOptions($options) tries to call AllowNull() method first with option's value as parameter, whereupon option will not be set.